spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java

```diff
@@ -317,31 +317,58 @@ public void testDirectWriteToBigQueryWithDiffInSchemaAndDisableModeCheck() throw
             "select * from %s.%s",
             testDataset.toString(), TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME);
     int numOfRows = (int) bq.query(QueryJobConfiguration.of(query)).getTotalRows();
-    assertThat(numOfRows).isEqualTo(1);
+    assertThat(numOfRows).isGreaterThan(0);
   }
 
   @Test
-  public void testInDirectWriteToBigQueryWithDiffInSchemaAndModeCheck() {
-    assumeThat(writeMethod, equalTo(SparkBigQueryConfig.WriteMethod.INDIRECT));
+  @Ignore("DSv2 only")
+  public void testDirectWriteToBigQueryWithDiffInDescription() throws Exception {
+    assumeThat(writeMethod, equalTo(WriteMethod.DIRECT));
     spark.conf().set("temporaryGcsBucket", temporaryGcsBucket);
     Dataset<Row> df =
         spark
             .read()
             .format("bigquery")
-            .option("table", testDataset + "." + TestConstants.DIFF_IN_SCHEMA_SRC_TABLE_NAME)
+            .option(
+                "table",
+                testDataset + "." + TestConstants.DIFF_IN_SCHEMA_SRC_TABLE_NAME_WITH_DESCRIPTION)
             .load();
 
     assertThrows(
-        Exception.class,
+        ProvisionException.class,
         () ->
             df.write()
                 .format("bigquery")
                 .mode(SaveMode.Append)
                 .option("writeMethod", writeMethod.toString())
-                .option("enableModeCheckForSchemaFields", true)
                 .save(testDataset + "." + TestConstants.DIFF_IN_SCHEMA_DEST_TABLE_NAME));
   }
 
+  @Test
+  public void testInDirectWriteToBigQueryWithDiffInSchemaAndModeCheck() throws Exception {
+    assumeThat(writeMethod, equalTo(SparkBigQueryConfig.WriteMethod.INDIRECT));
+    spark.conf().set("temporaryGcsBucket", temporaryGcsBucket);
```

Comment:
don't set globally, put in the write options

Reply:
done